### PR TITLE
Parallize workload driver inter-node communication

### PR DESCRIPTION
**Pull request by hrosenbe**
This pull request includes the following changes:
- Parallelize sending of messages among workload driver nodes.  This decreases the time required to make transitions between load-path intervals.
- Keep all workloads (appInstances) in a run (workload) in sync during the run (applies to all sub-types of syncedLoadPath).  Previously a workload could start the next interval before others.
- Make collecting csv stats and per-target stats optional.  This reduces overhead and storage requirements on the run harness pod.
- Add better logging of errors if the data preparation process fails.

### CODE REVIEW

File: runHarness/DataManagers/AuctionKubernetesDataManager.pm [hunk 1]
```diff
@@ -138,6 +139,25 @@ sub stopDataManagerContainer {
 	my $logger         = get_logger("Weathervane::DataManager::AuctionKubernetesDataManager");
 	my $cluster = $self->host;
 	
+	# before deleting the deployment, capture info about state of the datamanage
```
Comment (bhoflich) [inline runHarness/DataManagers/AuctionKubernetesDataManager.pm:142]: datamanage -> datamanager

File: runHarness/DataManagers/AuctionKubernetesDataManager.pm [hunk 2]
```diff
@@ -138,6 +139,25 @@ sub stopDataManagerContainer {
...
+	# pod and dump it to the log
+	my $namespace = $self->appInstance->namespace;
+	my $kubernetesConfigFile = $cluster->getParamValue('kubeconfigFile');
+	my $context = $cluster->getParamValue('kubeconfigContext');
+	my $contextString = "";
+	if ($context) {
+	  $contextString = "--context=$context";	
+	}
+	my $cmd = "kubectl describe pod --namespace=$namespace --selector=impl=auctiondatamanager --kubeconfig=$kubernetesConfigFile $contextString";
+    my ($cmdFailed, $outString) = runCmd($cmd);
```
Comment (bhoflich) [inline runHarness/DataManagers/AuctionKubernetesDataManager.pm:152]: The start line formatting varies through this section down through line 176, if that matters.

**Review by bhoflich — APPROVED** [review]

File: runHarness/DataManagers/AuctionKubernetesDataManager.pm (hunk 1 above)
Comment (hrosenbe) [reply runHarness/DataManagers/AuctionKubernetesDataManager.pm]: Fixed

File: runHarness/DataManagers/AuctionKubernetesDataManager.pm (hunk 2 above)
Comment (hrosenbe) [reply runHarness/DataManagers/AuctionKubernetesDataManager.pm]: fixed

**Pull request merged**